### PR TITLE
mgmt: Add support for un-registering a group along with the same support in img_mgmt

### PR DESCRIPTION
**Pull request by vrahane**
- Add support for un-registering a group at runtime 
- Add support for un-registering an image management group at runtime
- Re-entrancy to be handled by caller since the API modifies a list

### CODE REVIEW

**Review by mlaz — APPROVED** [review]
LGTM

**Comment by sjanc** [comment]
https://travis-ci.org/github/apache/mynewt-nimble/jobs/758517372

this breaks build...

**Comment by utzig** [comment]
> https://travis-ci.org/github/apache/mynewt-nimble/jobs/758517372
> 
> this breaks build...

About time CI is added to this repo...

File: mgmt/src/mgmt.c [hunk 1]
```diff
@@ -71,6 +71,31 @@ mgmt_streamer_free_buf(struct mgmt_streamer *streamer, void *buf)
     streamer->cfg->free_buf(buf, streamer->cb_arg);
 }
 
+void
+mgmt_unregister_group(struct mgmt_group *group)
+{
+    struct mgmt_group *curr = mgmt_group_list, *prev;
+
+    if (curr && curr == group) {
```
Comment (utzig) [inline mgmt/src/mgmt.c:79]: There is no need to test `curr` here because if it is NULL it won't be equal to `group`

File: mgmt/src/mgmt.c [hunk 2]
```diff
@@ -71,6 +71,31 @@ mgmt_streamer_free_buf(struct mgmt_streamer *streamer, void *buf)
     streamer->cfg->free_buf(buf, streamer->cb_arg);
 }
 
+void
+mgmt_unregister_group(struct mgmt_group *group)
+{
+    struct mgmt_group *curr = mgmt_group_list, *prev;
```
Comment (utzig) [inline mgmt/src/mgmt.c:77]: Please check that `group` is not NULL

File: mgmt/src/mgmt.c [hunk 3]
```diff
@@ -71,6 +71,31 @@ mgmt_streamer_free_buf(struct mgmt_streamer *streamer, void *buf)
...
+
+    if (curr && curr == group) {
+        mgmt_group_list = curr->mg_next;
+        return;
+    }
+
+    while (curr && curr != group) {
+        prev = curr;
```
Comment (utzig) [inline mgmt/src/mgmt.c:85]: If `curr` is valid and equal to `group`, `prev` will remain uninitialized.

**Review by utzig — COMMENTED** [review]
@vrahane Please fix the issues in a new PR.

File: mgmt/src/mgmt.c (hunk 2 above)
Comment (vrahane) [reply mgmt/src/mgmt.c:77]: Happy to do that, the only reason I did not add that is because mgmt_reguster_group() does not do it either.

File: mgmt/src/mgmt.c (hunk 1 above)
Comment (vrahane) [reply mgmt/src/mgmt.c:79]: Yes, agreed.

Comment (vrahane) [reply mgmt/src/mgmt.c:79]: The only reason I keep checking current everywhere is just to be safe since there was no NULL check for group.

File: mgmt/src/mgmt.c (hunk 3 above)
Comment (vrahane) [reply mgmt/src/mgmt.c:85]: Agreed, this needs a fix.

**Comment by vrahane** [comment]
> https://travis-ci.org/github/apache/mynewt-nimble/jobs/758517372
> 
> this breaks build...

Thank you @sjanc, I did not see any build issues on the zephyr build locally. @utzig ’s PR will fix it.

File: mgmt/src/mgmt.c (hunk 3 above)
Comment (vrahane) [reply mgmt/src/mgmt.c:85]: curr will never be equal to group here because we take care of that condition above.

File: mgmt/src/mgmt.c (hunk 2 above)
Comment (vrahane) [reply mgmt/src/mgmt.c:77]: I think I know the reason why the NULL check is not there, it’s because memory for the groups is not allocated here, it is a structure defined in the static memory that does not change. Nonetheless, does not hurt to add the check here.
